### PR TITLE
♻️ Leverage 'Point' struct on inventory functions

### PR DESCRIPTION
**Pull request by julealgon**
This PR replaces usages of individual X and Y coordinates in `inv.h`/`inv.cpp` functions with the `Point` structure.

### CODE REVIEW

File: Source/inv.cpp [hunk 1]
```diff
@@ -1228,10 +1228,10 @@ void CheckInvCut(int pnum, int mx, int my, bool automaticMove)
 		}
 
 		// check which inventory rectangle the mouse is in, if any
-		if (mx >= InvRect[r].X + xo
-		    && mx < InvRect[r].X + xo + (INV_SLOT_SIZE_PX + 1)
-		    && my >= InvRect[r].Y + yo - (INV_SLOT_SIZE_PX + 1)
-		    && my < InvRect[r].Y + yo) {
+		if (cursorPosition.x >= InvRect[r].X + xo
```
Comment (AJenbo) [inline Source/inv.cpp:1231]: On a side note, we would probably replace `InvXY` with `Point`

File: Source/inv.cpp [hunk 2]
```diff
@@ -1430,9 +1430,9 @@ void inv_update_rem_item(int pnum, BYTE iv)
 void CheckInvItem(bool isShiftHeld)
 {
 	if (pcurs >= CURSOR_FIRSTITEM) {
-		CheckInvPaste(myplr, MouseX, MouseY);
+		CheckInvPaste(myplr, { MouseX, MouseY });
```
Comment (AJenbo) [inline Source/inv.cpp:1433]: Looking forwared to you converting `MouseX/Y` and `ViewX/Y` with `Point`.

File: Source/inv.cpp [hunk 3]
```diff
@@ -1766,22 +1766,19 @@ bool TryInvPut()
...
 	}
 
-	position = myPlayer.position.tile + right[dir];
-	if (CanPut(position.x, position.y)) {
+	if (CanPut(myPlayer.position.tile + right[dir])) {
 		return true;
 	}
 
-	return CanPut(myPlayer.position.tile.x, myPlayer.position.tile.y);
+	return CanPut(myPlayer.position.tile);
```
Comment (AJenbo) [inline Source/inv.cpp:1781]: :tada:

**Review by AJenbo — APPROVED** [review]

File: Source/inv.cpp (hunk 1 above)
Comment (julealgon) [reply Source/inv.cpp:1231]: Agreed.

File: Source/inv.cpp (hunk 2 above)
Comment (julealgon) [reply Source/inv.cpp:1433]: Yep.... once the more basic replacements are done, I'll start looking into these shared globals and arrays.

**Pull request merged**